File: app/src/main/java/com/relic/data/gateway/SubGateway.java
package com.relic.data.gateway;

import android.arch.lifecycle.LiveData;

public interface SubGateway {
  LiveData<String> getAdditionalSubInfo(String subredditName);
  LiveData<Boolean> getIsSubscribed(String subredditName);

  LiveData<Boolean> subscribe(String subreddit);
  LiveData<Boolean> unsubscribe(String subreddit);

  void retrieveSubBanner(String subreddit);
}

File: app/src/main/java/com/relic/data/gateway/SubGateway.kt
package com.relic.data.gateway

interface SubGateway {
    suspend fun retrieveAdditionalSubInfo(subredditName: String): String

    suspend fun retrieveSidebar(subredditName: String): String

    suspend fun getIsSubscribed(subredditName: String): Boolean

    suspend fun subscribe(isSubbed: Boolean, subreddit: String)

    suspend fun retrieveSubBanner(subreddit: String)
}
